mne/io/base.py

```diff
@@ -202,6 +202,50 @@ def to_data_frame(self, picks=None, index=None, scale_time=1e3,
         return df
 
 
+class TimeMixin(object):
+    """Class to add sfreq and time_as_index capabilities to certain classes."""
+
+    def time_as_index(self, times, use_first_samp=None, use_rounding=False):
```

Comment:
I don't really like the fact that a use_first_samp param is exposed for non raw objects.